scripts/version.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/python3
+
+import argparse
+import json
+import os
+
+modules_to_show_in_version = ["mono"]
+
+
+def __make_version_header(json_version_file: str, out_version_file: str, out_hash_file: str, built_modules: [str]):
+    version = {}
+    with open(json_version_file, "r") as f:
+        version = json.load(f)
+
+    build_name = "custom_build"
+    if os.getenv("BUILD_NAME") != None:
+        build_name = os.getenv("BUILD_NAME")
+        print("Using custom build name: " + build_name)
+
+    version_modules = [x for x in built_modules if x in modules_to_show_in_version]
+
+    f = open(out_version_file, "w")
+    f.write("/* THIS FILE IS GENERATED DO NOT EDIT */\n")
+    f.write("#ifndef VERSION_GENERATED_GEN_H\n")
+    f.write("#define VERSION_GENERATED_GEN_H\n")
+    f.write('#define VERSION_SHORT_NAME "' + str(version["short_name"]) + '"\n')
+    f.write('#define VERSION_NAME "' + str(version["name"]) + '"\n')
+    f.write("#define VERSION_MAJOR " + str(version["major"]) + "\n")
+    f.write("#define VERSION_MINOR " + str(version["minor"]) + "\n")
+    f.write("#define VERSION_PATCH " + str(version["patch"]) + "\n")
+    f.write('#define VERSION_STATUS "' + str(version["status"]) + '"\n')
+    f.write('#define VERSION_BUILD "' + str(build_name) + '"\n')
+
+    version_array = [str(version["module_config"])] + version_modules
+
+    f.write('#define VERSION_MODULE_CONFIG "' + ".".join(version_array) + '"\n')
+    f.write("#define VERSION_YEAR " + str(version["year"]) + "\n")
+    f.write('#define VERSION_WEBSITE "' + str(version["website"]) + '"\n')
+    f.write("#endif // VERSION_GENERATED_GEN_H\n")
+    f.close()
```

Comment:
Never use explicit `close` in such situations!!!! You must use `with open(out_version_file, "w") as f:` context.